misc/natural_sort.c

```diff
@@ -51,9 +52,9 @@ int mp_natural_sort_cmp(const char *name1, const char *name2)
                 name2++;
             }
         } else {
-            if (mp_tolower(name1[0]) < mp_tolower(name2[0]))
+            if (mp_tolower_unsigned(name1[0]) < mp_tolower_unsigned(name2[0]))
                 return -1;
-            if (mp_tolower(name1[0]) > mp_tolower(name2[0]))
+            if (mp_tolower_unsigned(name1[0]) > mp_tolower_unsigned(name2[0]))
```

Comment:
Would it not be simpler to just cast the result or mp_tolower to unsignhed char, instead of adding a function which does exactly that (AFAICT) and only used in one place?

Reply:
Thanks. LGTM.

Reply:
Actually wait. Since name1 and name2 are now unsigned char*,  I don't think the cast is still required?

Reply:
You're right. I thought mp_tolower was a macro.

But then, I don't think we need the type change of name1 and name2, because eventhough chars in name1 and name2 are compared also regardless of mp_tolower, it only happens when both are digits, so [signed] char is fine in this case. Right?

Reply:
Thanks. I like tiny PRs ;)

LGTM.